### DISCUSSION TITLE
ApplicationContext.ServiceProviderScope in Csla 5.4.0

### DISCUSSION THREAD

**Discussion by ajohnstone-ks** [MarimerLLC/csla#1991] [General]
It looks like the type of ApplicationContext.ServiceProviderScope was changed from IServiceScope to IServiceProvider.  That seems like a mistake.  If not, it doesn't seem there's anyway now to set a scope to use.

**Comment by rockfordlhotka** [comment]
Yes, that does seem like a mistake. I think (not sure) that came with the service provider changes from @dazinator - any insight on this?

**Reply by dazinator** [reply]
No it was intentional,
When you create a service scope, the IServiceScope has an IServiceProvider exposed as a property - that's the thing that CSLA needs. You can hand CSLA your root application IServiceProvider, or you can create scopes and hand CSLA your scopes IServiceProvider, csla doesn't care, it will use the IServiceProvider provided to do its DI.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.iservicescope?view=dotnet-plat-ext-5.0

**Reply by ajohnstone-ks** [reply]
Ok, can I suggest that property get renamed to something else then?  Its a bit confusing because there is a scope concept as well.

**Reply by rockfordlhotka** [reply]
Would you create an issue for this please?

**Reply by ajohnstone-ks** [reply]
Oops, when I got the email with your response I thought it was about the other discussion.  

This is capture here:  https://github.com/MarimerLLC/csla/issues/1999
